src/main/java/sevensmurfs/rehub/service/TherapyResultService.java

```diff
@@ -28,4 +32,24 @@ public void deleteAllTherapyResultsForTherapies(List<Therapy> therapies) {
 
         log.debug("Successfully deleted all therapy results.");
     }
+
+    @Transactional
+    public void writeTherapyResult(TherapyResultRequest request) {
+        // Find the therapy by ID
+        Therapy therapy = therapyRepository.findById(request.getTherapyId())
+                                           .orElseThrow(() -> new EntityNotFoundException("Therapy not found"));
+
+        // Create a new TherapyResult entity
+        TherapyResult therapyResult = new TherapyResult();

```

Comment:
Replace constructor with builder. If it is not available add @Builder to TherapyResult class with @All and @NoArgsConstructor.